Add support for Python 3.13

The build is currently failing on macOS:

```
Compiling [macosx-shared] modules/IcePy/ValueFactoryManager.cpp
modules/IcePy/Util.cpp:619:12: error: use of undeclared identifier 'PyEval_CallObject'
  619 |     return PyEval_CallObject(type, args.get());
      |            ^
modules/IcePy/Util.cpp:654:16: error: use of undeclared identifier 'PyEval_CallObject'
  654 |         return PyEval_CallObject(type, pArgs.get());
      |                ^
2 errors generated.
make: *** [modules/IcePy/build/macosx/obj/Util.o] Error 1
make: *** Waiting for unfinished jobs....
modules/IcePy/Types.cpp:306:23: error: use of undeclared identifier 'PyEval_CallObject'
  306 |     PyObjectHandle sd{PyEval_CallObject(_slicedDataType, args.get())};
      |                       ^
modules/IcePy/Types.cpp:332:30: error: use of undeclared identifier 'PyEval_CallObject'
  332 |         PyObjectHandle slice{PyEval_CallObject(_sliceInfoType, args.get())};
      |                              ^
modules/IcePy/Types.cpp:1480:21: error: use of undeclared identifier 'PyObject_AsReadBuffer'; did you mean 'PyObject_GetBuffer'?
 1480 |                 if (PyObject_AsReadBuffer(p, &buf, &sz) == 0)
      |                     ^~~~~~~~~~~~~~~~~~~~~
      |                     PyObject_GetBuffer
/Users/joe/.local/share/mise/installs/python/latest/include/python3.13/pybuffer.h:46:17: note: 'PyObject_GetBuffer' declared here
   46 | PyAPI_FUNC(int) PyObject_GetBuffer(PyObject *obj, Py_buffer *view,
      |                 ^
modules/IcePy/Types.cpp:1480:46: error: cannot initialize a parameter of type 'Py_buffer *' with an rvalue of type 'const void **'
 1480 |                 if (PyObject_AsReadBuffer(p, &buf, &sz) == 0)
      |                                              ^~~~
/Users/joe/.local/share/mise/installs/python/latest/include/python3.13/pybuffer.h:46:62: note: passing argument to parameter 'view' here
   46 | PyAPI_FUNC(int) PyObject_GetBuffer(PyObject *obj, Py_buffer *view,
      |                                                              ^
4 errors generated.
make: *** [modules/IcePy/build/macosx/obj/Types.o] Error 1
```

--- a/python/modules/IcePy/Types.cpp
+++ b/python/modules/IcePy/Types.cpp
@@ -303,7 +303,7 @@ IcePy::StreamUtil::setSlicedDataMember(PyObject* obj, const Ice::SlicedDataPtr& 
         throw AbortMarshaling();
     }
 
-    PyObjectHandle sd{PyEval_CallObject(_slicedDataType, args.get())};
+    PyObjectHandle sd{PyObject_CallObject(_slicedDataType, args.get())};
     if (!sd.get())
     {
         assert(PyErr_Occurred());
@@ -329,7 +329,7 @@ IcePy::StreamUtil::setSlicedDataMember(PyObject* obj, const Ice::SlicedDataPtr& 
     int i = 0;
     for (vector<Ice::SliceInfoPtr>::const_iterator p = slicedData->slices.begin(); p != slicedData->slices.end(); ++p)
     {
-        PyObjectHandle slice{PyEval_CallObject(_sliceInfoType, args.get())};
+        PyObjectHandle slice{PyObject_CallObject(_sliceInfoType, args.get())};
         if (!slice.get())
         {
             assert(PyErr_Occurred());
@@ -1476,18 +1476,19 @@ IcePy::SequenceInfo::marshal(
             Py_ssize_t sz = 0;
             if (p != Py_None)
             {
-                const void* buf = 0;
-                if (PyObject_AsReadBuffer(p, &buf, &sz) == 0)
+                Py_buffer pybuf;
+                if (PyObject_GetBuffer(p, &pybuf, PyBUF_SIMPLE | PyBUF_FORMAT) == 0)
                 {
                     if (pi->kind == PrimitiveInfo::KindString)
                     {
                         PyErr_Format(PyExc_ValueError, "expected sequence value");
                         throw AbortMarshaling();
                     }
+                    sz = pybuf.len;
                 }
                 else
                 {
-                    PyErr_Clear(); // PyObject_AsReadBuffer sets an exception on failure.
+                    PyErr_Clear(); // PyObject_GetBuffer sets an exception on failure.
 
                     PyObjectHandle fs;
                     if (pi)
@@ -1711,7 +1712,7 @@ IcePy::SequenceInfo::marshalPrimitiveSequence(const PrimitiveInfoPtr& pi, PyObje
     if (pi->kind != PrimitiveInfo::KindString)
     {
         //
-        // With Python 3 and greater we marshal sequences of pritive types using the new
+        // With Python 3 and greater we marshal sequences of primitive types using the new
         // buffer protocol when possible, for older versions we use the old buffer protocol.
         //
         Py_buffer pybuf;
@@ -1820,7 +1821,7 @@ IcePy::SequenceInfo::marshalPrimitiveSequence(const PrimitiveInfoPtr& pi, PyObje
         }
         else
         {
-            PyErr_Clear(); // PyObject_GetBuffer/PyObject_AsReadBuffer sets an exception on failure.
+            PyErr_Clear(); // PyObject_GetBuffer sets an exception on failure.
         }
     }
 

--- a/python/modules/IcePy/Types.cpp
+++ b/python/modules/IcePy/Types.cpp
@@ -296,14 +296,7 @@ IcePy::StreamUtil::setSlicedDataMember(PyObject* obj, const Ice::SlicedDataPtr& 
         assert(_sliceInfoType);
     }
 
-    IcePy::PyObjectHandle args{PyTuple_New(0)};
-    if (!args.get())
-    {
-        assert(PyErr_Occurred());
-        throw AbortMarshaling();
-    }
-
-    PyObjectHandle sd{PyObject_CallObject(_slicedDataType, args.get())};
+    PyObjectHandle sd{PyObject_CallObject(_slicedDataType, nullptr)};
     if (!sd.get())
     {
         assert(PyErr_Occurred());
@@ -329,7 +322,7 @@ IcePy::StreamUtil::setSlicedDataMember(PyObject* obj, const Ice::SlicedDataPtr& 
     int i = 0;
     for (vector<Ice::SliceInfoPtr>::const_iterator p = slicedData->slices.begin(); p != slicedData->slices.end(); ++p)
     {
-        PyObjectHandle slice{PyObject_CallObject(_sliceInfoType, args.get())};
+        PyObjectHandle slice{PyObject_CallObject(_sliceInfoType, nullptr)};
         if (!slice.get())
         {
             assert(PyErr_Occurred());
@@ -1477,14 +1470,12 @@ IcePy::SequenceInfo::marshal(
             if (p != Py_None)
             {
                 Py_buffer pybuf;
-                if (PyObject_GetBuffer(p, &pybuf, PyBUF_SIMPLE | PyBUF_FORMAT) == 0)
+                if (pi && PyObject_GetBuffer(p, &pybuf, PyBUF_SIMPLE | PyBUF_FORMAT) == 0)
                 {
-                    if (pi->kind == PrimitiveInfo::KindString)
-                    {
-                        PyErr_Format(PyExc_ValueError, "expected sequence value");
-                        throw AbortMarshaling();
-                    }
+                    // Strings are handled as variable length types above.
+                    assert(pi->kind != PrimitiveInfo::KindString);
                     sz = pybuf.len;
+                    PyBuffer_Release(&pybuf);
                 }
                 else
                 {

--- a/python/modules/IcePy/Util.cpp
+++ b/python/modules/IcePy/Util.cpp
@@ -616,7 +616,7 @@ IcePy::createExceptionInstance(PyObject* type)
     {
         return nullptr;
     }
-    return PyEval_CallObject(type, args.get());
+    return PyObject_CallObject(type, args.get());
 }
 
 namespace
@@ -651,7 +651,7 @@ namespace
             // PyTuple_SetItem takes ownership of the args[i] reference.
             PyTuple_SetItem(pArgs.get(), static_cast<Py_ssize_t>(i), args[i]);
         }
-        return PyEval_CallObject(type, pArgs.get());
+        return PyObject_CallObject(type, pArgs.get());
     }
 }
 

--- a/python/modules/IcePy/Util.cpp
+++ b/python/modules/IcePy/Util.cpp
@@ -611,12 +611,7 @@ PyObject*
 IcePy::createExceptionInstance(PyObject* type)
 {
     assert(PyExceptionClass_Check(type));
-    IcePy::PyObjectHandle args{PyTuple_New(0)};
-    if (!args.get())
-    {
-        return nullptr;
-    }
-    return PyObject_CallObject(type, args.get());
+    return PyObject_CallObject(type, nullptr);
 }
 
 namespace

